core/src/main/java/com/graphhopper/routing/ev/SignedDecimalEncodedValue.java

```diff
@@ -0,0 +1,117 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.ev;
+
+import com.graphhopper.storage.IntsRef;
+
+/**
+ * This class holds a decimal value and stores it as an unsigned integer value via a conversion factor and a maximum number
+ * of bits.
+ */
+public final class SignedDecimalEncodedValue extends SignedIntEncodedValue implements DecimalEncodedValue {
+    private final double factor;
+    private final boolean defaultIsInfinity;
```

Comment:
I did not look at this in detail yet, but would it be possible to just set an arbitrary value as default, not just choose between 0 and infinity?

Reply:
Yes, we do this in master. But we do not use this flexibility (we only use 0 and infinity). And setting the default value will then also be possible via setting the value to 0, which is not intuitive as you expect that getInt(ref) returns 0 if you set it to 0 before. Even with infinity we have a problem and reject values with 0 (see last two commits) because there seems to be osm tags with e.g. `width=0`.

Probably we should try remove the default handing and set this in the TagParser only. But probably in a later commit.

Reply:
> Probably we should try remove the default handing and set this in the TagParser only. But probably in a later commit.

Ok, yes. I'd like to assume that each TagParser sets a value for every edge, and if it does not this is an error and we must expect a 'garbage' value. The only other reasonable expectation would be that it is zero when nothing was set, but only because initially all bits of the underlying `int[]` or `byte[]` are zero. However, this would no longer work if we implemented a 'mapped' encoded value where 0 is mapped to something else, so let's just say not setting a value for an edge is an error.

Reply:
> Ok, yes. I'd like to assume that each TagParser sets a value for every edge

I thought a bit about this and currently for some EncodedValues in FlagEncoders we might skip the set call due to getAccess returning "SKIP". So a proper default was/is required in general and I'll keep the default mechanics for now. But we should probably work towards a default-free version of EncodedValues, i.e. ensuring every EncodedValue is called for every edge on import.

Reply:
Yes, I meant we should aim for this, but currently this would require further changes that are beyond the scope of this PR. Also #2462 will help with this I think.